src/js/releases.js

```diff
@@ -40,77 +35,62 @@ function buildLatestHTML(releasesJson) {
-      }
-      // if the filename contains both the platform name and the matching BINARY extension, add the relevant info to the asset object
-      if (uppercaseFilename.indexOf(ASSETOBJECT.thisBinaryExtension.toUpperCase()) >= 0) {
-        var installerExist = false;
-        if (ASSETARRAY.length > 0) {
-          ASSETARRAY.forEach(function (asset) {
-            if (asset.thisPlatform === ASSETOBJECT.thisPlatform) {
-              installerExist = true;
-            }
-          });
-        }
-        if (!installerExist) {
-          ASSETOBJECT.thisPlatformExists = true;
-          ASSETOBJECT.thisBinaryExists = true;
-          ASSETOBJECT.thisBinaryLink = eachAsset.binary_link;
-          ASSETOBJECT.thisBinarySize = Math.floor(eachAsset.binary_size / 1024 / 1024);
-          ASSETOBJECT.thisChecksumLink = eachAsset.checksum_link
-        }
-      }
+  // Array of releases that have binaries we want to display
+  var releases = [];
 
-      if (ASSETOBJECT.thisPlatformExists === true) {
-        ASSETARRAY.push(ASSETOBJECT);
-      }
+  releasesJson.forEach((releaseAsset) => {
+    const platform = findPlatform(releaseAsset);
+
+    // Skip this asset if its platform could not be matched (see the website's 'config.json')
+    if (!platform) {
+      return;
+    }
+
+    // Skip this asset if it's not a binary type we're interested in displaying
+    const binary_type = releaseAsset.binary_type.toUpperCase();
+    if (['INSTALLER', 'JDK', 'JRE'].indexOf(binary_type) === -1) {
+      return;
+    }
+
+    // Get the existing release asset (passed to the template) or define a new one
+    var release = releases.find((p) => p.platform_name === platform);
+    if (!release) {
+      release = {
+        platform_name: platform,
+        platform_official_name: getOfficialName(platform),
+        platform_ordinal: getPlatformOrder(platform),
+        platform_logo: getLogo(platform),
+
+        release_name: releaseAsset.release_name,
+        release_link: releaseAsset.release_link,
+        release_datetime: moment(releaseAsset.timestamp).format('YYYY-MM-DD hh:mm:ss'),
+
+        binaries: []
+      };
+    }
+
+    // Add the new binary to the release asset
+    release.binaries.push({
+      type: binary_type,
+      extension: 'INSTALLER' === binary_type ? getInstallerExt(platform) : getBinaryExt(platform),
+      link: releaseAsset.binary_link,
+      checksum_link: releaseAsset.checksum_link,
+      size: Math.floor(releaseAsset.binary_size / 1024 / 1024)
+    });
 
+    // We have the first binary, so add the release asset.
+    if (release.binaries.length === 1) {
+      releases.push(release);
     }
   });
 
-  ASSETARRAY = orderPlatforms(ASSETARRAY);
+  releases = orderPlatforms(releases, 'platform_ordinal');
+  releases.forEach((r) => { r.binaries.sort((a, b) => a.type > b.type ? 1 : a.type < b.type ? -1 : 0) });
```

Comment:
`r` -> `release`, I'm also a fan of naming a and b in sort functions for the things they really are but YMMV

Reply:
Updated in ca6da266238294642fbb69949f7d568a4c38ff2c